src/cargo/core/compiler/fingerprint/mod.rs

```diff
@@ -1741,35 +1744,67 @@ fn compare_old_fingerprint(
     new_fingerprint: &Fingerprint,
     mtime_on_use: bool,
     forced: bool,
+    verbosity: Verbosity,
 ) -> Option<DirtyReason> {
-    let compare = _compare_old_fingerprint(old_hash_path, new_fingerprint, mtime_on_use);
-    log_compare(unit, &compare);
+    if mtime_on_use {
```

Comment:
The other commits seem nice and could be merged on their own.  I feel like this adds complexity that, without a motivating reason (showing a noticeable difference on end-to-end benchmarks), seems of questionable value.

Reply:
Sound reasonable. I was refactoring unnecessary Options and found it could be skipped. Will remove this part later.